src/java.base/share/classes/jdk/internal/reflect/Reflection.java

```diff
@@ -111,7 +111,13 @@ public static void ensureMemberAccess(Class<?> currentClass,
 
     @ForceInline
     public static void ensureNativeAccess(Class<?> currentClass, Class<?> owner, String methodName) {
-        Module module = currentClass.getModule();
+        class Holder {
+            private final static Module FALLBACK_MODULE =
+                    SharedSecrets.getJavaLangAccess().defineUnnamedModule(null); // lazy init
+        }
+        // if there is no caller class, act as if the call came from an unnamed module
+        Module module = currentClass != null ?
+                currentClass.getModule() : Holder.FALLBACK_MODULE;
```

Comment:
This can be simplified to:
```
Module module = currentClass != null ? currentClass.getModule() 
                                                              : ClassLoader::getSystemClassLoader().getUnnamedModule();
```

Reply:
gah! I missed that :-)